src/services/refactors/moveToFile.ts

```diff
@@ -0,0 +1,198 @@
+        return [...prologueDirectives, ...toMove.all];
+    }
+
+    const useEsModuleSyntax = !!oldFile.externalModuleIndicator;
+    const quotePreference = getQuotePreference(oldFile, preferences);
+    const importsFromNewFile = createOldFileImportsFromNewFile(oldFile, usage.oldFileImportsFromNewFile, newFilename, program, host, useEsModuleSyntax, quotePreference);
+    if (importsFromNewFile) {
+        insertImports(changes, oldFile, importsFromNewFile, /*blankLineBetween*/ true, preferences);
+    }
+
+    deleteUnusedOldImports(oldFile, toMove.all, changes, usage.unusedImportsFromOldFile, checker);
+    deleteMovedStatements(oldFile, toMove.ranges, changes);
+    updateImportsInOtherFiles(changes, program, host, oldFile, usage.movedSymbols, newFilename, quotePreference);
+
+    const imports = getNewFileImportsAndAddExportInOldFile(oldFile, newFile, usage.oldImportsNeededByNewFile, usage.newFileImportsFromOldFile, changes, checker, program, host, useEsModuleSyntax, quotePreference, importAdder);
+    const body = addExports(oldFile, toMove.all, usage.oldFileImportsFromNewFile, useEsModuleSyntax);
+    if (newFileExists) {
+        const sourceFile = program.getSourceFile(newFile);
+        if (sourceFile && sourceFile.statements.length > 0) {
+            changes.insertNodesAfter(sourceFile, sourceFile.statements[sourceFile.statements.length - 1], body);
+        }
+        if (sourceFile && imports.length > 0) {
+            insertImports(changes, sourceFile, imports, /*blankLineBetween*/ true, preferences);
+        }
+    }
+    if (importAdder) {
+        importAdder.writeFixes(changes);
+    }
+    if (imports.length && body.length) {
+        return [
+            ...prologueDirectives,
+            ...imports,
+            SyntaxKind.NewLineTrivia as const,
+            ...body
+        ];
+    }
+
+    return [
+        ...prologueDirectives,
+        ...imports,
+        ...body,
+    ];
+}
+
+function getNewFileImportsAndAddExportInOldFile(
+    oldFile: SourceFile,
+    newFile: string,
+    importsToCopy: Map<Symbol, boolean>,
+    newFileImportsFromOldFile: Set<Symbol>,
+    changes: textChanges.ChangeTracker,
+    checker: TypeChecker,
+    program: Program,
+    host: LanguageServiceHost,
+    useEsModuleSyntax: boolean,
+    quotePreference: QuotePreference,
+    importAdder?: codefix.ImportAdder,
+): readonly AnyImportOrRequireStatement[] {
+    const copiedOldImports: AnyImportOrRequireStatement[] = [];
+    if (importAdder) {
+        importsToCopy.forEach((isValidTypeOnlyUseSite, symbol) => {
+            importAdder.addImportFromExportedSymbol(skipAlias(symbol, checker), isValidTypeOnlyUseSite);
+        });
+    }
+    else {
+        for (const oldStatement of oldFile.statements) {
+            forEachImportInStatement(oldStatement, i => {
+                append(copiedOldImports, filterImport(i, moduleSpecifierFromImport(i), name => importsToCopy.has(checker.getSymbolAtLocation(name)!)));

```

Comment:
It looks like this assumes the module specifier in the import in the old file will work in the new file, which isn’t correct. Consider moving an import `import { a } from "./a"` into a new file that’s not a sibling of the old file. The module specifier might need to be rewritten to something like `"./subdir/a"` or `"../../a"` depending on the relationship between the locations of the source and target files.